engine/render/src/render/font_renderer.cpp

```diff
@@ -898,6 +897,9 @@ namespace dmRender
                         continue;
                     }
 
+                    // If we get here, then it may be that c != glyph->m_Character
+                    c = g->m_Character;
```

Comment:
Old behavior:
If the glyph doesn't exist, it checks if 126 `~` exists

So the fix is to simply use the codepoint of the glyph we retrieved (and not the one we originally  requested)